hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/HawkbitUIErrorHandler.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.components;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.hawkbit.ui.utils.HawkbitCommonUtil;
+import org.eclipse.hawkbit.ui.utils.I18N;
+import org.eclipse.hawkbit.ui.utils.SpringContextHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.server.DefaultErrorHandler;
+import com.vaadin.server.ErrorEvent;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.themes.ValoTheme;
+
+/**
+ * Default handler for SP UI.
+ */
+public class HawkbitUIErrorHandler extends DefaultErrorHandler {
+
+    /**
+     * Comment for <code>serialVersionUID</code>.
+     */
+    private static final long serialVersionUID = 1877326479308824191L;
+    /**
+     * logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(HawkbitUIErrorHandler.class);
+
+    private HawkbitNotificationMessage notification;
+    private StringBuilder style;
+    private I18N i18n;
+    private String exceptionName;
+    private String exceptionMessage;
+    protected Component errorOrigin;
+
+    @Override
+    public void error(final ErrorEvent event) {
+        notification = new HawkbitNotificationMessage();
+        // Build error style
+        style = new StringBuilder(ValoTheme.NOTIFICATION_FAILURE).append(' ').append(ValoTheme.NOTIFICATION_SMALL)
+                .append(' ').append(ValoTheme.NOTIFICATION_CLOSABLE);
+        i18n = SpringContextHelper.getBean(I18N.class);
+        // From the exception trace we get the expected exception class name
+        if (StringUtils.isBlank(exceptionName)) {
+            exceptionName = getExceptionName(event);
+        }
+        if (getAndCheckErrorOrigin(event)) {
+            showNotification();
+        }
+    }
+
+    /**
+     * Determines and returns the name of the thrown exception. Stores the
+     * exception message.
+     * 
+     * @param event
+     *            ErrorEvent
+     * @return name of the thrown exception
+     */
+    protected String getExceptionName(final ErrorEvent event) {
+        for (Throwable error = event.getThrowable(); error != null; error = error.getCause()) {
+            exceptionName = HawkbitCommonUtil.getLastSequenceBySplitByDot(error.getClass().getName());
+            exceptionMessage = error.getMessage();
+            LOG.error("Error in SP-UI:", error);
+        }
+        return exceptionName;
```

Comment:
You do not return the name from `ErrorEvent` every time! So why I have to provide it for the method call?